### DISCUSSION TITLE
Very Good Idea

### DISCUSSION THREAD

**Discussion by ssfve** [finol-digital/Card-Game-Simulator#38] [Ideas]
It's a very good idea. I think if this is a simulator, there is already a lot of customization (for creators to use and share their decks and make a digital counterpart of all current card game on the market). Currently, there is already a lot digital card game on the market like hearthstone, but not much thing like a customized card game ready to play. The simulator will be a great tool for card game designers to do testing before a card game conceives.
But if we want many more others to use the app(for a lot other users who can't create and won't create), I think we should make the app more like playing cards in a real environment for example at home(it should be a **SEPARATE** app, just **SEPARATE** the code). We open a deck of cards in the app, open the card box, put on card sleeves maybe, then play a game under **LAN** (which is crucial because such a game must be played face to face), just like a traditional card game played at home on a game night. After separation, the two apps could still connect, card deck created in one app will be asset which could be downloaded into the other app for playing purpose. 
Then we will be replacing millions and millions of card decks produced in the world now. We will be replacing or at least copying all materialized paper cards and traditional decks into digital ones at a very low cost (playing cards on a phone just like holding cards in your hands). That could be really great changes in the card and printing industry in my opinion.

**Comment by davidmfinol** [comment]
I agree that it is critical to be able to "open a deck of cards in the app, open the card box, put on card sleeves maybe, then play a game under LAN (which is crucial because such a game must be played face to face), just like a traditional card game played at home on a game night".

I'm currently focusing on improving the experience of playing the cards for users that aren't interested in creating new games. 
Truthfully, even though I want to also support creating new games and cards directly within the app, that functionality is a secondary priority that is still a long way away from working the way I want it to.

Why have the game creation functionality exist in a separate app?
Do you think people might get distracted or confused by the ability to create and share games? 
I think it would be better to have it all in 1 app, since that would make it easier for people that want to create new card games and share it with other people. 
Personally, I'm envisioning a scenario like YouTube: where >99% of users are consumers of content created by <1% of users. And even though the creators make up such a small % of the userbase, the ability to create & upload is heavily promoted and accessible.

**Comment by ssfve** [comment]
Do you think people might get distracted or confused by the ability to create and share games?
Yes, I agree creating and sharing should be put in 1 app for creators, but after many contents are created, for content consumers, I think that should be in a separate app. The two apps are still in a ecosystem, they work together, each serves their purposes. OR if there is a config to hide the creator's functionality, just like developer's mode in android, then it looks just like a different app for the specific user in my opinion.

**Reply by davidmfinol** [reply]
I want to encourage being able to easily create content and share it, and I think having it all be in a single app will help with that. 
You do raise valid concerns about users potentially being confused by the creation functionality, so I think the best approach is like you suggested: a config that enables access to a "developer mode". And this config would be be off by default, of course.
For now, I'll continue to focus on improving the play experience, but I'll be sure to add this config when I am able to work on the create &share functionality.

**Reply by ssfve** [reply]
I am just expressing my concerns. It does not mean you need to do anything immediately. There are surely more important things to do. I sincerely support the idea of card simulator and hope it will become better.

**Comment by Bendell90** [comment]
I'm sure you're familiar with it but have you tried out Table Top Simulator? Alot of what's going on there could help a bunch with setting the baseline for interaction, customization, etc.
Personally I think card games could be mobile in distance (IMO legit not trying to be rude) its definitely something that would benefit from voice chat atleast though. 

But regardless I wonder what's your major goal with this game?

**Reply by davidmfinol** [reply]
Yep, I do take a lot of inspiration from TableTop Simulator!
I do have goals similar to TableTop Simulator, but I think mobile is a big use case for me.

Card games are the kind of thing that are more fun when physically in the same place, but you can't always carry cards on you.
People do always carry their phone on them, so the goal is to provide a platform for people to just play whatever card game they want, whenever they want, without having to bring physical cards (just the phones that they already have).

That said, I do think having an in-game voice chat option would be good for purely virtual game sessions, and it's something that I would also like to eventually support.
